File: Interpreted/lcs-recursive/lcs.jl
function lcs(xstr, ystr)

    (xstr == "" || ystr == "") && return "";

    x = xstr[1:1];
    xs = xstr[2:end];
    y = ystr[1:1];
    ys = ystr[2:end];

    x == y && return(x * lcs(xs, ys));

    lcs1 = lcs(xstr, ys);
    lcs2 = lcs(xs, ystr);

    length(lcs1) > length(lcs2) ? lcs1 : lcs2;
end

println(lcs(ARGS[1], ARGS[2]));

File: Interpreted/lcs-recursive/lcs.rb
def lcs(xstr, ystr)

    xstr.empty? || ystr.empty? and return "";

    x = xstr[0];
    xs = xstr[1..-1];
    y = ystr[0];
    ys = ystr[1..-1];

    x == y and return(x + lcs(xs, ys));

    lcs1 = lcs(xstr, ys);
    lcs2 = lcs(xs, ystr);

    lcs1.size > lcs2.size ? lcs1 : lcs2;
end

puts lcs(ARGV[0], ARGV[1]);
